sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp

```diff
@@ -123,6 +123,17 @@ namespace Azure { namespace Storage { namespace Blobs {
    */
   struct BlobClientOptions
   {
+    /**
+     * Versioning for Azure Storage Services.
+     */
+    struct ServiceVersion
+    {
+      /**
+       * 2020-02-10 service version.
+       */
+      constexpr static const char* V2020_02_10 = "2020-02-10";
+    };
```

Comment:
Shouldn't this be an enum/extensible enum?

In which case, should this be a class with `const static` values?
https://azure.github.io/azure-sdk/cpp_implementation.html#enumerations

cc @JeffreyRichter

Reply:
Extensible enum is used when client side needs to parse a value returned from server side. I don't see when this SDK or customer needs to do that. Actually I don't even know when this service version string will be used.

Reply:
I thought this version was going to be passed as the API version header to the service, but might be mistaken. @tg-msft, are we supposed to plumb the `ServiceVersion` string to any of the storage service API calls?

Reply:
Jinming is correct; this should not be an extensible enum.
I think we just want a string field in the XxxClient options struct called ApiVersion. By default, we set this to the api-version that corresponds to the latest version of the service and this version is what we test & support.
We want to allow the customer to pass their own string if they want to throw caution to the wind and just try an arbitrary api-version. Usually, the customers experience will be fine if they do this but, we need to say that this is not a tested & supported scenario. This enables scenarios like Azure Stack Hub and other non-public clouds to work that do not have the latest version of the service available.

Reply:
Aside from the `ApiVersion` string field, why not help such a customer by providing them with the list of valid api version strings to choose from, rather than let them pick any arbitrary free-form string?

This way, let's say we ship storage v2, we will have V2020_02_10 and say V2021_12_31, and then the user can set `ApiVersion` to either one of those (still best effort), rather than the user having to fish out from storage service docs the version strings.

Reply:
Because we don't support this. It feels odd to me to here are a bunch of possible values and we don't guarantee any of them will work for you.

Reply:
Hi @JeffreyRichter , are you suggesting that I should remove what I've added in this PR and add an extra `std::string` field into `BlobClientOptions`, the default value for this `std::string` should be the latest API version we support, while we still allow our customers to change this `std::string` value if they want to try another API version?

Reply:
Yes, exactly.

Reply:
@JeffreyRichter Thanks for your clarification!